### PR TITLE
Input validation added to create a service

### PR DESCRIPTION
**Pull request by ramakantar**
- Input validation for creating an inventory
- Input validation for creating a user
- Input validation for creating a group
- Input validation for creating a project

### CODE REVIEW

File: awx.go [hunk 1]
```diff
@@ -5,6 +5,9 @@ import (
 	"net/http"
 )
 
+// This variable is to be populated for creating services
```
Comment (Colstuwjx) [inline awx.go]: maybe `mandatory fields for services API`?

Comment (ramakantar) [reply awx.go]: You're right. Fixed it.

File: users.go [hunk 2]
```diff
@@ -34,6 +35,14 @@ func (u *UserService) ListUsers(params map[string]string) ([]*User, *ListUsersRe
 
 // CreateUser creates an awx User.
 func (u *UserService) CreateUser(data map[string]interface{}, params map[string]string) (*User, error) {
+	mandatoryFields = []string{"username", "password", "first_name", "last_name", "email"}
```
Comment (Colstuwjx) [inline users.go]: ONLY the username is required, see https://docs.ansible.com/ansible-tower/3.2.2/html/towerapi/users.html#reviewing-the-options-endpoint

File: projects.go [hunk 3]
```diff
@@ -34,6 +35,14 @@ func (p *ProjectService) ListProjects(params map[string]string) ([]*Project, *Li
 
 // CreateProject creates an awx project.
 func (p *ProjectService) CreateProject(data map[string]interface{}, params map[string]string) (*Project, error) {
+	mandatoryFields = []string{"name", "organization", "scm_type"}
```
Comment (Colstuwjx) [inline projects.go]: ONLY `name` is required, see https://docs.ansible.com/ansible-tower/latest/html/towerapi/projects.html#reviewing-the-options-endpoint

Comment (ramakantar) [reply projects.go]: I followed the doc but received 400. So I checked in gui.
screen shot 2018-08-28 at 16 45 55

Comment (Colstuwjx) [reply projects.go]: Okay, let's just merge, and let this as another PR's task!

Comment (Colstuwjx) [reply projects.go]: Okay, let's just merge and leave this as another PR's task.

**Comment by Colstuwjx** [comment]
Thanks.

**Pull request merged**